models/iaf_cond_alpha_mc_kinetics.cpp

```diff
@@ -0,0 +1,875 @@
+  updateValue< double >( d, names::tau_h, tau_h );
+  updateValue< double >( d, names::half_m, half_m );
+  updateValue< double >( d, names::half_h, half_h );
+  updateValue< double >( d, names::slope_m, slope_m );
+  updateValue< double >( d, names::slope_h, slope_h );
+  updateValue< double >( d, names::jump_Th, jump_Th );
+  updateValue< double >( d, names::tau_Th, tau_Th );
+  updateValue< double >( d, names::act_flag, act_flag );
+  updateValue< double >( d, names::reset_flag, reset_flag );
+  updateValue< double >( d, Name( "g_sp" ), g_conn[ SOMA ] );
+  updateValue< double >( d, Name( "g_pd" ), g_conn[ PROX ] );
+
+  // extract from sub-dictionaries
+  for ( size_t n = 0; n < NCOMP; ++n )
+    if ( d->known( comp_names_[ n ] ) )
+    {
+      DictionaryDatum dd = getValue< DictionaryDatum >( d, comp_names_[ n ] );
+
+      updateValue< double >( dd, names::t_L, t_L[ n ] );
+      updateValue< double >( dd, names::nt_L, nt_L[ n ] );
+      updateValue< double >( dd, names::E_L, E_L[ n ] );
+      updateValue< double >( dd, names::E_ex, E_ex[ n ] );
+      updateValue< double >( dd, names::E_in, E_in[ n ] );
+      updateValue< double >( dd, names::C_m, C_m[ n ] );
+      updateValue< double >( dd, names::g_L, g_L[ n ] );
+      updateValue< double >( dd, names::tau_syn_ex, tau_synE[ n ] );
+      updateValue< double >( dd, names::tau_syn_in, tau_synI[ n ] );
+      updateValue< double >( dd, names::I_e, I_e[ n ] );
+      // changes
+      updateValue< double >( dd, names::tau_cur_AP, tau_curAP[ n ] );
+      updateValue< double >( dd, names::amp_cur_AP, amp_curAP[ n ] );
+    }
+
+  if ( V_reset >= V_th )
+    throw BadProperty( "Reset potential must be smaller than threshold." );
+
+  if ( t_ref < 0 )
+    throw BadProperty( "Refractory time cannot be negative." );
+
+  if ( tau_m <= 0 || tau_h <= 0 || tau_Th <= 0 )
+    throw BadProperty( "All time constants must be strictly positive." );
+
+  // apply checks compartment-wise
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    if ( C_m[ n ] <= 0 )
+      throw BadProperty(
+        "Capacitance (" + comp_names_[ n ].toString() + ") must be strictly positive." );
+
+    if ( tau_synE[ n ] <= 0 || tau_synI[ n ] <= 0 || tau_curAP[ n ] <= 0 )
+      throw BadProperty(
+        "All time constants (" + comp_names_[ n ].toString() + ") must be strictly positive." );
+  }
+}
+
+void
+nest::iaf_cond_alpha_mc_kinetics::State_::get( DictionaryDatum& d ) const
+{
+  // we assume here that State_::get() always is called after Parameters_::get(),
+  // so that the per-compartment dictionaries exist
+  for ( size_t n = 0; n < NCOMP; ++n )
+  {
+    assert( d->known( comp_names_[ n ] ) );
+    DictionaryDatum dd = getValue< DictionaryDatum >( d, comp_names_[ n ] );
+
+    def< double >( dd, names::V_m, y_[ idx( n, V_M ) ] ); // Membrane potential
+  }
+}
+
+void
+nest::iaf_cond_alpha_mc_kinetics::State_::set( const DictionaryDatum& d, const Parameters_& )
```

Comment:
there is a getca function which can be called in pynest to get the calcium current values.